firecracker/kernel/build.sh

```diff
@@ -13,35 +13,54 @@ readonly REPOSITORY_ROOT
 
 BUILD_DIR="$(mktemp --directory)"
 readonly BUILD_DIR
-cd "${BUILD_DIR}"
 
 ########################################
 # Get the Firecracker source
 ########################################
-git clone git@github.com:firecracker-microvm/firecracker.git \
-    --depth=1 \
-    --branch="${FIRECRACKER_RELEASE}"
-cd firecracker
+(
+    cd "${BUILD_DIR}"
+    git clone git@github.com:firecracker-microvm/firecracker.git \
+        --depth=1 \
+        --branch="${FIRECRACKER_RELEASE}"
+    cd firecracker
 
-########################################
-# Generate the kernel.
-# Based on https://github.com/firecracker-microvm/firecracker/blob/main/docs/rootfs-and-kernel-setup.md
-########################################
-./tools/devtool build_kernel --config "${KERNEL_CONFIG}" --nproc 8
+    ########################################
+    # Generate the kernel.
+    # Based on https://github.com/firecracker-microvm/firecracker/blob/main/docs/rootfs-and-kernel-setup.md
+    ########################################
+    ./tools/devtool build_kernel --config "${KERNEL_CONFIG}" --nproc 8
+)
 
 ########################################
 # Copy kernel into dist.
 ########################################
 readonly KERNEL_BIN_DIR="${BUILD_DIR}/firecracker/build/kernel/linux-${KERNEL_VERSION}/"
 readonly KERNEL_BIN_FILE="vmlinux-${KERNEL_VERSION}-x86_64.bin"
-readonly DISTRIBUTION="${REPOSITORY_ROOT}/dist/firecracker_kernel.tar.gz"
+readonly ARTIFACT_PATH="${REPOSITORY_ROOT}/dist/firecracker_kernel.tar.gz"
 
 # NOTE about tar: If you specify the full path of the thing-to-be-tar'd,
 #   the tar will contain that full nested path of directories.
 #   Hence the --directory, and basename for KERNEL_BIN_FILE
 tar \
     --directory "${KERNEL_BIN_DIR}" \
-    --file="${DISTRIBUTION}" \
+    --file="${ARTIFACT_PATH}" \
     --create \
     --gzip \
     "${KERNEL_BIN_FILE}"
+
+########################################
+# Write a .artifact-metadata.json file
+########################################
+source .buildkite/scripts/lib/artifact_metadata.sh
+ARTIFACT_METADATA_PATH="$(artifact_metadata_path "${ARTIFACT_PATH}")"
+readonly ARTIFACT_METADATA_PATH
+
+source .buildkite/scripts/lib/version.sh
+# Note: in Cloudsmith, versions must be len <=128, hence we use a short git sha.
+VERSION="firecracker-${FIRECRACKER_RELEASE}-kernel-${KERNEL_VERSION}-$(sha_version)"
+readonly VERSION
+
+TAG="inputsha-$(sha256_of_dir firecracker/kernel)"
```

Comment:
Nit: I think `input-sha256:${THE_SHA256}` would be a better format:

- it is a bit more readable (to me, at least: hyphens rather than all the words smooshed together)
- the `:` suggests a key-value pair, whereas the `-` doesn't
- "input-sha256" tells you exactly what type of checksum it is, whereas "input-sha" leaves you guessing.